src/main/java/com/mraof/minestuck/client/gui/SendificatorScreen.java

```diff
@@ -0,0 +1,150 @@
+			this.destinationTextFieldX.setValue(String.valueOf(te.getDestinationBlockPos().getX()));
+			addButton(destinationTextFieldX);
+			
+			this.destinationTextFieldY = new TextFieldWidget(this.font, this.width / 2 + 25, 50, 20, 15, new StringTextComponent("Y value of destination block pos"));
+			this.destinationTextFieldY.setValue(String.valueOf(te.getDestinationBlockPos().getY()));
+			addButton(destinationTextFieldY);
+			
+			this.destinationTextFieldZ = new TextFieldWidget(this.font, this.width / 2 + 45, 50, 35, 15, new StringTextComponent("Z value of destination block pos"));
+			this.destinationTextFieldZ.setValue(String.valueOf(te.getDestinationBlockPos().getZ()));
+			addButton(destinationTextFieldZ);
+			
+			addButton(new ExtendedButton((width - imageWidth) / 2 + 105, 80, 50, 12, new StringTextComponent("Update"), button -> updateDestinationPos()));
+		}
+		
+		//activates processContents() in SendificatorTileEntity
+		goButton = new GoButton((width - imageWidth) / 2 + goX, (height - imageHeight) / 2 + goY, 30, 12, new StringTextComponent(menu.overrideStop() ? "STOP" : "GO"));
+		addButton(goButton);
+	}
+	
+	@Override
+	public void render(MatrixStack matrixStack, int mouseX, int mouseY, float partialTicks)
+	{
+		this.renderBackground(matrixStack);
+		super.render(matrixStack, mouseX, mouseY, partialTicks);
+		this.renderTooltip(matrixStack, mouseX, mouseY);
+	}
+	
+	@Override
+	protected void renderLabels(MatrixStack matrixStack, int mouseX, int mouseY)
+	{
+		//draws the name of the TE
+		font.draw(matrixStack, this.title.getString(), 8, 6, 4210752);
+		
+		//draws "Inventory" or your regional equivalent
+		font.draw(matrixStack, this.inventory.getDisplayName().getString(), 8, imageHeight - 96 + 2, 4210752);
+	}
+	
+	@Override
+	protected void renderBg(MatrixStack matrixStack, float par1, int par2, int par3)
+	{
+		RenderSystem.color4f(1.0F, 1.0F, 1.0F, 1.0F);
+		
+		//draw background
+		this.minecraft.getTextureManager().bind(BACKGROUND);
+		int x = (width - imageWidth) / 2;
+		int y = (height - imageHeight) / 2;
+		this.blit(matrixStack, x, y, 0, 0, imageWidth, imageHeight);
+		
+		//draw progress bar
+		this.minecraft.getTextureManager().bind(PROGRESS);
+		int width = progressWidth;
+		int height = getScaledValue(menu.getFuel(), SendificatorTileEntity.getMaxFuel(), progressHeight);
+		blit(matrixStack, x + progressX, y + progressY + progressHeight - height, 0, progressHeight - height, width, height, progressWidth, progressHeight);
+	}
+	
+	private void updateDestinationPos()
+	{
+		if(te != null)
+			MSPacketHandler.sendToServer(new SendificatorPacket(parseBlockPos(), te.getBlockPos()));
+	}
+	
+	private static int parseInt(TextFieldWidget widget)
+	{
+		try
+		{
+			return Integer.parseInt(widget.getValue());
+		} catch(NumberFormatException ignored)
+		{
+			return 0;
```

Comment:
Consider expanding on the destination update handling by:
- Not accepting invalid text instead of interpreting invalid text as 0
- Have the update button be inactive until the destination coordinates have changed
- Perhaps even be inactive in general when inputted destination matches the currently set destination
The first point would ideally involve letting the `NumberFormatException` pass through `parseInt()` and `parseBlockPos()`, to instead be handled by the caller to `parseBlockPos()`.
If you need to listen to changes in the text field text, you can use `textField.setResponder()` after creating the text field to set a listener.

Reply:
Changes I just added should be fine as we looked over the majority of it together, but will give you a quick moment before considering this truly resolved